mmv1/templates/terraform/decoders/tags_tag_binding.go.erb

```diff
@@ -0,0 +1,34 @@
+<%# The license inside this block applies to this file.
+	# Copyright 2021 Google Inc.
+	# Licensed under the Apache License, Version 2.0 (the "License");
+	# you may not use this file except in compliance with the License.
+	# You may obtain a copy of the License at
+	#
+	#     http://www.apache.org/licenses/LICENSE-2.0
+	#
+	# Unless required by applicable law or agreed to in writing, software
+	# distributed under the License is distributed on an "AS IS" BASIS,
+	# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	# See the License for the specific language governing permissions and
+	# limitations under the License.
+-%>
+// func resourceTagBindingDecoder(d *schema.ResourceData, meta interface{}, res map[string]interface{}) (map[string]interface{}, error)
```

Comment:
We're going to want to handle this using a `nested_query` instead of just a decoder, I think. That may? give us a chance at handling pagination, too.

Reply:
I did try using `nested_query`, but it didn't quite work right for me. Specifically, this line: https://github.com/GoogleCloudPlatform/magic-modules/blob/master/mmv1/templates/terraform/nested_query.go.erb#L46

It resolved to:
```go
expectedFlattenedName := flattenNestedTagBindingName(expectedName, d, meta.(*Config))
```

Which resulted in `expectedName` (which was already correct) being re-processed, resulting in an incorrect "expected name".

But maybe I can work around it.

Reply:
What were your `identity` and `nested_query` values set to? Can you push a commit to this branch or another one including them?

It's been a while since I've worked with `nested_query` so seeing the generated code will help-- this is the kind of problem that it solves.

Reply:
So the issue is that flattenNestedTagBindingName needs to be idempotent - i.e. if you have a string like `this/is/a/long/id` and you run it through the default flattener, you're fine, because the default flatterer runs:

```
parts := strings.Split(link, "/")
return parts[len(parts)-1]
```

Which is idempotent because it always returns the last section of the /-separated string. But I was doing:

```
parts := strings.SplitN(v.(string), "/", 2)
return parts[1]
```

So if you run that on the same string more than once, you slowly chop off pieces from the beginning. I fixed it by making the return value idempotent by basing it on distance from the end of the string:

```
parts := strings.Split(v.(string), "/")
return strings.Join(parts[len(parts)-3:], "/")
```

Resolved by ac7fee2